moderator/moderate/auth.py

```diff
@@ -1,84 +1,55 @@
 from django.conf import settings
 from django.contrib.auth.models import User
+from django.core.exceptions import SuspiciousOperation
 
 from mozilla_django_oidc.auth import OIDCAuthenticationBackend
 
-from moderator.moderate.mozillians import BadStatusCode, MozilliansClient, ResourceDoesNotExist
-
 
 class ModeratorAuthBackend(OIDCAuthenticationBackend):
     """Override base authentication class."""
 
-    def __init__(self, *args, **kwargs):
-        """Add the mozillian user in the init method."""
-        self.mozillian_user = None
-        self.mozillians_client = MozilliansClient(settings.MOZILLIANS_API_URL,
-                                                  settings.MOZILLIANS_API_KEY)
+    def get_or_create_user(self, access_token, id_token, payload):
+        """Create a new user only if there is a vouched mozillians.org account."""
 
-        super(ModeratorAuthBackend, self).__init__(*args, **kwargs)
+        user_info = self.get_userinfo(access_token, id_token, payload)
 
-    def create_user(self, claims, **kwargs):
-        """Create a new user only if there is a vouched mozillians.org account."""
+        email = user_info.get("email")
+        groups = user_info.get("https://sso.mozilla.com/claim/groups", [])
 
-        email = claims.get('email')
-        try:
-            self.mozillian_user = self.mozillians_client.lookup_user({'email': email})
-        except (BadStatusCode, ResourceDoesNotExist):
+        claims_verified = self.verify_claims(user_info)
+        if not claims_verified:
+            msg = "Claims verification failed"
+            raise SuspiciousOperation(msg)
+
+        # The user is not staff or NDA member. Return None
+        if not any(x in groups for x in settings.ALLOWED_LOGIN_GROUPS):
             return None
 
-        user_emails = []
-        if self.mozillian_user['is_vouched']:
-            for email_entry in self.mozillian_user['alternate_emails']:
-                user_emails.append(email_entry['email'])
-            user_emails.append(self.mozillian_user['email']['value'])
-            users = User.objects.filter(email__in=user_emails)
-            if users:
-                return users[0]
-            return super(ModeratorAuthBackend, self).create_user(claims, **kwargs)
+        # email based filtering
```

Comment:
Is there a reason we copy the behaviour from `OIDCAuthenticationBackend` here, rather than calling `super`? Is it to avoid calling `get_userinfo` twice? If so, we could override that to return a cached result if it's already been called, something like:
```
def get_userinfo():
    if not self.cached_userinfo:
        self.cached_userinfo = super()
    return self.cached_userinfo
```

Reply:
The reason for moving the logic below the `if not any()` is that we don't want to either return but mostly create a user if the `access group` is missing from the claims. If we can super we will create an entry for that user in the DB and then return none b/c the user is missing the right groups.

Reply:
Couldn't we still do that filtering and then call super? Just replacing L28 onwards with the super call:
```
user_info = self.get_userinfo()

...

if not any(x in groups ...):
    return None

return super()
```